pkg/controller/cluster/main-controller.go

```diff
@@ -490,6 +490,35 @@ func (mk *minioKeychain) Resolve(_ authn.Resource) (authn.Authenticator, error)
 	}), nil
 }
 
+// getKeychainForTenant attempts to build a new authn.Keychain from the image pull secret on the Tenant
+func (c *Controller) getKeychainForTenant(ref name.Reference, tenant *miniov1.Tenant) (authn.Keychain, error) {
+	// Get the secret
+	secret, err := c.kubeClientSet.CoreV1().Secrets(tenant.Namespace).Get(context.Background(), tenant.Spec.ImagePullSecret.Name, metav1.GetOptions{})
+	if err != nil {
+		return nil, errors.New("can't retrieve the tenant image pull secret")
+	}
+	// if we can't find .dockerconfigjson, error out
+	if _, ok := secret.Data[".dockerconfigjson"]; !ok {
+		return nil, errors.New("can't find .dockerconfigjson in image pull secret")
+	}
+	var config configfile.ConfigFile
+	err = json.Unmarshal(secret.Data[".dockerconfigjson"], &config)
```

Comment:
```suggestion
	err = json.Unmarshal(secret.Data[".dockerconfigjson"], &config)
```

Avoid double map lookups instead 

```
dockerConfigJSON, ok := secret.Data[".dockerconfigjson"]; 
if !ok {
}
...
err = json.Unmarshal(dockerConfigJSON, &config)
```